scipy/stats/_axis_nan_policy.py

```diff
@@ -289,14 +356,16 @@ def axis_nan_policy_wrapper(*args, _no_deco=False, **kwds):
             contains_nan, _ = (
                 scipy.stats._stats_py._contains_nan(x, nan_policy))
 
-            if vectorized and not contains_nan:
+            if vectorized and not contains_nan and not sentinel:
                 return hypotest_fun_in(*samples, axis=axis, **kwds)
 
             # Addresses nan_policy == "omit"
             if contains_nan and nan_policy == 'omit':
                 def hypotest_fun(x):
                     samples = np.split(x, split_indices)[:n_samp]
                     samples = _remove_nans(samples, paired)
+                    if sentinel:
+                        samples = _remove_sentinel(samples, paired, sentinel)
```

Comment:
![image](https://user-images.githubusercontent.com/43181252/146652082-6aeb91c0-e241-47bf-a7fa-6e7abf125ae9.png)

Not sure why these lines aren't covered.

Reply:
Thanks for taking a look. I'll make sure this gets hit by a test.

Reply:
Fixed in latest commit.